### DISCUSSION TITLE
Recursive `mapEntries()`

### DISCUSSION THREAD

**Discussion by RobinBobin** [radashi-org/radashi#329] [Ideas]
Hi everyone,
I need to map snake case keys to camel case in objects I get from a backend. As far as I understand, `mapEntries()` doesn't go deep into nested values. I implemented this functionality, so I wanted to ask whether it's worth making a PR.
Thanks.

**Comment by aleclarson** [comment]
I think it's best if we gather more use cases first.

In your case, the implementation wouldn't need circular reference detection, since database results typically don't include circular references, whereas a general solution would need that detection. Re-casing keys from database results is typically a performance-sensitive task, since database retrieval is commonly a hot path of SaaS APIS, for example. Therefore, you wouldn't want to waste CPU cycles on circular reference detection and the general solution would be less than ideal.

There's also a question of how to handle arrays and `Map` objects.
- Do array indices get mapped by the callback, or are they only recursed in search of plain objects? In my opinion, it should be the latter, but some use cases may prefer the former.
- Same question for `Map` objects. In your case, it'd be best performance-wise to not have any `Map` handling, since database retrieval never returns them.

If we do add such a thing, I'd suggest the following behavior:
- Assume JSON data only (no handling of `Map`, `Set`, class-based objects)
- The callback only maps key-value pairs of plain objects
- Arrays are recursed, but only copied if a plain object is found within
- No support for circular references

**Comment by RobinBobin** [comment]
I totally agree :thumbs_up: .
Actually I get JSON from my backend, so I didn’t even think of any other possibilities :-) .
